include/EVT/dev/encoder.hpp

```diff
@@ -0,0 +1,108 @@
+#ifndef _EVT_ENCODER_
+#define _EVT_ENCODER_
+
+#include <EVT/io/GPIO.hpp>
+#include <EVT/utils/log.hpp>
+#include <EVT/utils/time.hpp>
+
+namespace time = EVT::core::time;
+
+namespace EVT::core::DEV {
+
+/**
+ * This class represents a CTS Series 288 Rotary encoder with a 2-bit Binary resolution.
+ * The static wrappers for the interrupt handles must be created in the main method
+ * see samples/encoder/main.cpp for example implementation.
+ * update() must be called within the main loop to update encoder position.
+ */
+class Encoder {
+public:
+    /**
+     * Constructs an Encoder instance.
+     *
+     * @param[in] a a pin of the encoder
+     * @param[in] b b pin of the encoder
+     * @param range range of the encoder positions
```

Comment:
Here and in the functions below
```suggestion
     * @param[in] range range of the encoder positions
```

Reply:
done!